packages/flutter/lib/src/rendering/stack.dart

```diff
@@ -781,7 +781,11 @@ class RenderIndexedStack extends RenderStack {
     while (child != null) {
       children.add(child.toDiagnosticsNode(
         name: 'child ${i + 1}',
-        style: i != index ? DiagnosticsTreeStyle.offstage : null,
+        // Though those children are indeed offstage, the current 
+        // `debugVisitOnstageChildren` wrongly treat them as onstage.
```

Comment:
This sounds like a bug in `debugVisitOnstageChildren`. Is there an issue for it in the tracker?

In general the Flutter approach to this situation is to strongly prefer fixing the underlying bug, rather than adding a workaround:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#lazy-programming

Reply:
Yes, that's definitely a bug on debugVisitOnstageChildren. However, someone has tried to fix that in https://github.com/flutter/flutter/pull/111479#issuecomment-1463754109, but that PR was closed because it was breaking change, and added to https://github.com/flutter/flutter/issues/24722.

Thus, I guess we have to do this PR if we do not want to confuse the users :/